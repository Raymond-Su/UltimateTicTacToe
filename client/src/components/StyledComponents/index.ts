export * from './styledButton';
export * from './styledContainer';
export * from './styledPanel';
