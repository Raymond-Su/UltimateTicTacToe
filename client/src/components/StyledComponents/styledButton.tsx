import Styled from 'styled-components';

interface styledButtonProps {
  primary?: boolean;
}

export const StyledButton = Styled.button<styledButtonProps>`
  display: inline-block;
  padding: 6px 12px;
  margin-bottom: 0;
  font-weight: 400;
  line-height: 1.42857143;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -ms-touch-action: manipulation;
  touch-action: manipulation;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-image: none;
  border: 1px solid #ccc;
  border-radius: 4px;
  color: ${(props) => (props.primary ? '#fff' : 'initial')};
  background-color: ${(props) => (props.primary ? '#337ab7' : 'initial')};
  border-color:  ${(props) => (props.primary ? '#2e6da4;' : 'initial')};
`;
