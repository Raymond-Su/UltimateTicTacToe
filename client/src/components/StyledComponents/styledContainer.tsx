import Styled from 'styled-components';

export const StyledContainer = Styled.div`
  width: 100%;
  padding-right: 10px;
  padding-left: 10px;
  margin-right: auto;
  margin-left: auto;
  box-sizing: border-box;

  @media (min-width: 992px) {
    min-width: 970px;
  }
`;

export const StyledResizeableLargeColumn = Styled.div`
  position: relative;
  min-height: 1px;
  padding-right: 15px;
  padding-left: 15px;
  box-sizing: border-box;

  @media (min-width: 992px) {
    width: 66.666667%;
    float: left;
    left: 33.33333333%;
`;

export const StyledResizeableSmallColumn = Styled.div`
  position: relative;
  min-height: 1px;
  padding-right: 15px;
  padding-left: 15px;
  box-sizing: border-box;
  
  @media (min-width: 992px) {
    width: 33.333333%;
    float: left;
    right: 66.66666667%;
  }
`;
