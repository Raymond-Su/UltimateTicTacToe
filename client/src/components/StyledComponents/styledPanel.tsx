import Styled from 'styled-components';

export const StyledPanel = Styled.div`
   margin-bottom: 20px;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 4px;
  -webkit-box-shadow: 0 1px 1px rgb(0 0 0 / 5%);
  box-shadow: 0 1px 1px rgb(0 0 0 / 5%);
`;

export const StyledPanelHeading = Styled.div`
  padding: 10px 15px;
  border-bottom: 1px solid transparent;
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
  color: #333;
  background-color: #f5f5f5;
  border-color: #ddd;
`;

export const StyledPanelTitle = Styled.h1`
  margin-top: 0;
  margin-bottom: 0;
  font-size: 16px;
  color: inherit;
`;

export const StyledPanelBody = Styled.div`
  padding: 15px;
  &::before {
    display: table;
    content: ' ';
  }
  &::after {
    clear: both;
  }
`;
