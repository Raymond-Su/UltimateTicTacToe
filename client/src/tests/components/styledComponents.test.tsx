import { render } from '@testing-library/react';
import React from 'react';
import { create } from 'react-test-renderer';

import { StyledButton } from '../../components/StyledComponents/styledButton';

const mockFunction = jest.fn();

describe('Styled components', () => {
  describe('<StyledButton />', () => {
    test('Renders component', () => {
      const { getByTestId } = render(<StyledButton data-testid="btn" />);
      const button = getByTestId('btn');
      expect(button).toBeInstanceOf(HTMLButtonElement);
    });

    test('primary props changes style', () => {
      const { getByTestId } = render(
        <StyledButton data-testid="btn" primary />
      );
      const button = getByTestId('btn');
      expect(button).toHaveStyle('color: #fff');
    });

    test('Matches the snapshot', () => {
      const button = create(
        <StyledButton onClick={mockFunction} primary id="snapshot-btn">
          Snapshot Button
        </StyledButton>
      );
      expect(button.toJSON()).toMatchSnapshot();
    });
  });
});
