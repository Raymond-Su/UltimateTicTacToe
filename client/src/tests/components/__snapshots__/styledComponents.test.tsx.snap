// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled components <StyledButton /> Matches the snapshot 1`] = `
<button
  className="sc-bdfBwQ jCToZW"
  id="snapshot-btn"
  onClick={[MockFunction]}
>
  Snapshot Button
</button>
`;
